### DISCUSSION TITLE
rpms lockfile maintainance

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#29578] [Suggest an Idea]
### Tell us more.

We at Red Hat, implemented a python package which maintains rpms lock file. 
https://github.com/konflux-ci/rpm-lockfile-prototype

Would it make sense to integrate it into the lockfile-maintainace function?

**Comment by rarkins** [comment]
In which manager would this go and what's the lock file name?

**Reply by lmilbaum** [reply]
Maybe a new manager?
The lock file name is: rpms.lock.yaml

**Reply by rarkins** [reply]
It's marked as unstable and a prototype, also had 1 star so far on GitHub. So it doesn't sound ready to add yet, if it requires us to add tooling to our docker images.

**Reply by lmilbaum** [reply]
That makes sense.  We already have a custom image baked with it. The challenge is to make it run as part of the lockfile maintainace function.  Are you aware of a configuration which can pick it up during runtime?

**Reply by rarkins** [reply]
Maybe postUpgradeTasks?

**Reply by lmilbaum** [reply]
That is already implemented.  IIUC, it triggers only if an existing dependency has changed. What is there are not existing dependencies besides the lockfile?

**Reply by rarkins** [reply]
I don't think we have a way of integration then. You'd need to run it as some type of scheduled pipeline which raises PRs if content changes

**Reply by lmilbaum** [reply]
Well, it is preferred to have one service which maintains our dependencies. The other option which was raised internally is to fork. That raises another set of concerns.
